packages/db/tests/models/test-file-model.js

```diff
@@ -105,3 +105,32 @@ test('FilePgModel.upsert() overwrites a file record', async (t) => {
     updatedFile
   );
 });
+
+test.serial('FilePgModel.upsert() creates a new file record if bucket/key are undefined', async (t) => {
+  const {
+    knex,
+    filePgModel,
+    granuleCumulusId,
+  } = t.context;
+
+  const file = fakeFileRecordFactory({
+    granule_cumulus_id: granuleCumulusId,
+    checksum_value: cryptoRandomString({ length: 3 }),
+    bucket: undefined,
+    key: undefined,
+  });
+  await filePgModel.create(knex, file);
+
+  const updatedFile = {
+    ...file,
+    checksum_value: cryptoRandomString({ length: 3 }),
+  };
+  await filePgModel.upsert(knex, updatedFile);
+
+  const records = await filePgModel.search(
+    knex,
+    { granule_cumulus_id: granuleCumulusId }
+  );
+
+  t.is(records.length, 2);
+});
```

Comment:
This test passes and I think that's a problem. If we try to upsert a file with an `undefined` bucket/key, Postgres will treat those values as unique and add a new row rather than updating any existing rows.

This SO post sums it up pretty well:

https://stackoverflow.com/questions/20154033/allow-null-in-unique-column

This might be an indication that the duplicate records concern that @markdboyd is still valid. To me it seems like a stronger argument for making bucket/key required in the File schema. Perhaps this is enough evidence that we can do it and communicate why the risk of possible failed writes was worth it? It makes that decision seem a little less abstract to me.

Reply:
Also note that this branch does not have the bucket or key as required columns. Also this test is run in `serial` to be sure that there are not other files in the table when we run it. If I remove the `upsert` on line 128 there is only one file in the table.

Reply:
Also x3: this test can stay I think but it's not related to skipping the file writes in the rest of this PR.

Reply:
Ah I think we talked about the null bucket/key thing when I was working on the initial implementation of granules and files migration because it was hard to look up duplicate records where the unique key was a kind of `null` key. 😬 

https://github.com/nasa/cumulus/pull/2051#issuecomment-763160033

Reply:
If it's a problem, we should probably write the failing case here.

Reply:
Yeah, I think it doesn't make sense to leave this test in. It causes some confusion now and that will compound as time goes by 😄 

We proved that allowing `undefined` buckets and keys will cause duplicate records to be created when doing an upsert. That's not the logic we want so having a test that proves it and passes is confusing. Also it's not really related to the changes in this PR.

This will be covered with better tests in https://github.com/nasa/cumulus/pull/2129